src/Functions/FunctionMathBinaryFloat64.h

```diff
@@ -53,35 +53,89 @@ class FunctionMathBinaryFloat64 : public IFunction
     }
 
     template <typename LeftType, typename RightType>
```

Comment:
If we do the conversion in advance by creating a temporary ColumnFloat64 with castColumn, the code will be simpler.

Reply:
This is done, please check.